### ISSUE TITLE
aws_ecs_task_definition overwrites previous revision

### ISSUE THREAD

**Issue by dimahavrylevych** [hashicorp/terraform#8740]
Hello community, 

I faced an issue while working with `aws_ecs_task_definition`. 
I have a script:

```
data "template_file" "task_definition" {
  template = "${file("task-definitions/service.json")}"
  vars {
    SERVICE_NAME = "test"
    IMAGE_URL = "test:latest"
  }
}

resource "aws_ecs_task_definition" "test" {
    family = "test"
    network_mode = "bridge"
    container_definitions = "${data.template_file.task_definition.rendered}"
}

resource "aws_ecs_service" "test" {
  name = "test"
  cluster = "test"
  task_definition = "${aws_ecs_task_definition.test.arn}"
  desired_count = 2
  deployment_minimum_healthy_percent = 100
}
```

Im trying to running: `terraform plan` so the part of output looks like:

```
-/+ ...
...
 revision:              "24" => "<computed>"
...
```

While running `terraform apply` and loging to AWS I see that the new revision has created but the previous one dissapeared. 
### Question:

Is is possible to implement a flag that will allow me to save previous revisions?

**Comment by radeksimko** [comment]
Hi,
this was originally (at the time of creation) intended behaviour to prevent Terraform from "leaking resources" (in this case registered task definitions), but I can see how that might confuse users.

See https://github.com/hashicorp/terraform/issues/8765 which should address this problem that affects couple of resources including ECS TD.
TL;DR after that change we'd be actually leaking resources, but we'd give the user fairly easy way to delete those instead of letting them do it outside of Terraform.

**Comment by dimahavrylevych** [comment]
ok, thanks @radeksimko 
do you guys have any progress on #8765 ?

**Comment by adamgotterer** [comment]
Is there a workaround that prevents aws_ecs_task_definition from creating a new task revision every time terraform apply is run?

**Comment by Dgadavin** [comment]
I think it will be useful to keep previous revision. Maybe I want to rollback after deploy.

**Comment by jim3ma** [comment]
+1

**Comment by madamedwards** [comment]
FWIW, previous revisions are not actually destroyed, they're just marked "inactive" in AWS. You can mark them active again manually if you need to roll back. However, it would still be great if the task definition resource had an option to leave the previous revision active.

**Comment by aholbreich** [comment]
How to workoround this?

**Comment by aholbreich** [comment]
> You can mark them active again manually if you need to roll back.

Didn't found the way to do that.. Can't do anythink with INACTIVE task-definition revisions. Only list.

**Comment by aderenbach** [comment]
A work around is to delete the terraform state before. Then a new resource is created and the old one is not getting inactive. 
Not a nice way, but I think the only solution for getting "versions"
